File: crud_java/src/main/java/com/example/crud_java/application/item/ItemService.java
package com.example.crud_java.application.item;

import com.example.crud_java.application.common.dto.PageDto;
import com.example.crud_java.application.item.dto.*;
import org.springframework.data.domain.Pageable;

public interface ItemService {
    PageDto readItems(Pageable pageable, ItemReadRequest request);
    Long createItem(ItemCreateRequest request);
    ItemReadResponse readItem(Long itemNo);
    void updateItem(Long itemNo, ItemUpdateRequest request);
    void updateItemQuantity(Long itemNo, ItemQuantityUpdateRequest request);
    void deleteItem(Long itemNo);
}

File: crud_kotlin/src/main/kotlin/com/example/crud_kotlin/application/item/ItemService.kt
package com.example.crud_kotlin.application.item

import com.example.crud_kotlin.application.common.dto.PageDto
import com.example.crud_kotlin.application.item.dto.*
import org.springframework.data.domain.Pageable

interface ItemService {
    fun readItems(pageable: Pageable, request: ItemReadRequest): PageDto<Iterable<ItemReadResponse>>
    fun createItem(request: ItemCreateRequest): Long
    fun readItem(itemNo: Long): ItemReadResponse
    fun updateItem(itemNo: Long, request: ItemUpdateRequest)
    fun updateItemQuantity(itemNo: Long, request: ItemUpdateQuantityRequest)
}